src/handshake/client.rs

```diff
@@ -160,7 +160,12 @@ impl VerifyData {
         // 1. If the status code received from the server is not 101, the
         // client handles the response per HTTP [RFC2616] procedures. (RFC 6455)
         if response.status() != StatusCode::SWITCHING_PROTOCOLS {
-            return Err(Error::Http(response.status()));
+            if response.status().is_redirection() {
+                let value = response.headers().get("Location").unwrap();
```

Comment:
While `Location` must be set in normal cases, we can't rely on that (someone could have implemented a bad web server that does not behave properly => we must not panic in such cases). I think the safer approach would be to check the existence of header.

Also we introduce a new variant on the `Error` for `Redirect`ing, but if I'm not missing on something, then it won't appear in the public API, i.e. no-one will ever be able to get this error as it's the one used only inside our handshake module which might be confusing for other people (they don't know if redirects are forwarded or not). Maybe we should still return an `Http` error like we did before, but instead of containing the `StatusCode` use the `Response`?

This would allow us to handle redirects at one place (if any).

Reply:
I'll fix the header checks in the next commit

If you prefer I can rebase when #150 is merged and build on that, or the changes there can be subsumed into this PR, makes no difference to me.

Reply:
The changes in #150 are minimal, I think we can do them right here.